### PR TITLE
[Bug] Handle the '\r\n' newline correctly

### PR DESCRIPTION
**Pull request by czechboy0**
### Motivation

Fixes https://github.com/apple/swift-openapi-generator/issues/143, which tracks including the '\r\n' newline in a description in the OpenAPI document.

### Modifications

Replaces the '\r\n' character with '\n' before rendering comments, as '\n' is handled properly.

### Result

OpenAPI documents that include '\r\n' in a description produce compiling code now.

### Test Plan

Added a unit test for this case.

### CODE REVIEW

**Comment by czechboy0** [comment]
@swift-server-bot test this please

File: Sources/_OpenAPIGeneratorCore/Renderer/TextBasedRenderer.swift [hunk 1]
```diff
@@ -54,12 +54,15 @@ struct TextBasedRenderer: RendererProtocol {
             prefix = "// MARK:"
             commentString = string
         }
-        return commentString.transformingLines { line in
-            if line.isEmpty {
-                return prefix
+        return
+            commentString
+            .replacingOccurrences(of: "\r", with: "\n")
```
Comment (simonjbeaumont) [inline Sources/_OpenAPIGeneratorCore/Renderer/TextBasedRenderer.swift:59]: In the presence of `\r\n` this produces two newlines but I think it should just be one.

Presumably the goal of this patch is to handle both newline conventions: `LF` or `CR+LF`. IIUC, the `\r` is a carriage return.

Does `String` have any utilities for splitting on "newlines" which might just handle this for us internally?

Comment (czechboy0) [reply Sources/_OpenAPIGeneratorCore/Renderer/TextBasedRenderer.swift:59]: Oh I wasn't sure, let me see if we can only produce a single newline here.

Comment (czechboy0) [reply Sources/_OpenAPIGeneratorCore/Renderer/TextBasedRenderer.swift]: Ok, how about this? https://github.com/apple/swift-openapi-generator/pull/169/commits/e2e7f3b2eaf42004e3e749bbcd046fa7d07b0cdc

File: Sources/_OpenAPIGeneratorCore/Renderer/TextBasedRenderer.swift [hunk 2]
```diff
@@ -721,7 +723,7 @@ fileprivate extension String {
     /// Returns an array of strings, where each string represents one line
     /// in the current string.
     func asLines() -> [String] {
-        split(separator: "\n", omittingEmptySubsequences: false)
+        split(omittingEmptySubsequences: false, whereSeparator: \.isNewline)
```
Comment (simonjbeaumont) [inline Sources/_OpenAPIGeneratorCore/Renderer/TextBasedRenderer.swift:726]: Should we be omitting empty sequences?

Realise this is scope creep because this was probably the behaviour before but I think that we should be preserving the number of newlines in comments. If you consider that a comment may be interpreted as markdown, you need two newlines to start a new block.

**Review by simonjbeaumont — APPROVED** [review]

File: Sources/_OpenAPIGeneratorCore/Renderer/TextBasedRenderer.swift (hunk 2 above)
Comment (czechboy0) [reply Sources/_OpenAPIGeneratorCore/Renderer/TextBasedRenderer.swift:726]: Yeah, this is existing behavior, and I don't remember all the reasons we explicitly added this. I suspect it was to preserve the newlines that adopters have in their comments, be it markdown or just plain text.

Comment (simonjbeaumont) [reply Sources/_OpenAPIGeneratorCore/Renderer/TextBasedRenderer.swift:726]: Oh. I completely inverted this check in my head! You’re right.
